src/tqec/templates/library/spatial.py

```diff
@@ -0,0 +1,430 @@
+    # contain any reset or measurement. It is up to the junction to use resets
+    # or measurements on the data-qubits shared if they are needed.
+    if JunctionArms.UP in arms and JunctionArms.LEFT in arms:
+        mapping[1] = RPNGDescription.from_string(f"---- -{be}2- -{be}4- -{be}5-")
+    if JunctionArms.DOWN in arms and JunctionArms.RIGHT in arms:
+        mapping[4] = RPNGDescription.from_string(f"-{be}1- -{be}2- -{be}4- ----")
+
+    ####################
+    #    Boundaries    #
+    ####################
+    # Fill the boundaries that should be filled in the returned template because
+    # they have no junction, and so will not be filled later.
+    # Note that indices 1 and 4 **might** be set twice in the 4 ifs below. These
+    # cases are handled later in the function and will overwrite the description
+    # on 1 and 4 if needed, so we do not have to account for those cases here.
+    # Also note that, by convention from the docstring, corners (so indices 1
+    # and 4 below) should NOT contain reset/measurement operations.
+    if JunctionArms.UP not in arms:
+        mapping[1] = RPNGDescription.from_string(f"---- ---- -{be}3- -{be}4-")
+        mapping[10] = RPNGDescription.from_string(f"---- ---- {r}{be}3{m} {r}{be}4{m}")
+    if JunctionArms.RIGHT not in arms:
+        mapping[4] = RPNGDescription.from_string(f"-{be}1- ---- -{be}2- ----")
+        mapping[18] = RPNGDescription.from_string(f"{r}{be}1{m} ---- {r}{be}2{m} ----")
+    if JunctionArms.DOWN not in arms:
+        mapping[4] = RPNGDescription.from_string(f"-{be}1- -{be}2- ---- ----")
+        mapping[20] = RPNGDescription.from_string(f"{r}{be}1{m} {r}{be}2{m} ---- ----")
+    if JunctionArms.LEFT not in arms:
+        mapping[1] = RPNGDescription.from_string(f"---- -{be}3- ---- -{be}4-")
+        mapping[12] = RPNGDescription.from_string(f"---- {r}{be}3{m} ---- {r}{be}4{m}")
+
+    # If we have a down-right or top-left L-shaped junction, the opposite corner
+    # plaquette should be removed from the mapping (this is the case where it
+    # has been set twice in the ifs above).
+    if arms == JunctionArms.UP | JunctionArms.LEFT:
+        del mapping[4]
+    elif arms == JunctionArms.DOWN | JunctionArms.RIGHT:
+        del mapping[1]
+
+    ####################
+    #       Bulk       #
+    ####################
+    # Assigning plaquette description to the bulk, considering that the bulk
+    # corners (i.e. indices {5, 6, 7, 8}) should be assigned "regular" plaquettes
+    # (i.e. 6 and 7 have the same plaquette as 17, 5 has the same plaquette as
+    # 13 and 8 has the same plaquette as 15). If these need to be changed, it
+    # will be done afterwards.
+    internal_basis_plaquette = RPNGDescription.from_string(
+        f"{r}{bi}1{m} {r}{bi}3{m} {r}{bi}2{m} {r}{bi}5{m}"
+    )
+    mapping[6] = internal_basis_plaquette
+    mapping[7] = internal_basis_plaquette
+    mapping[17] = internal_basis_plaquette
```

Comment:
At first, it was for future-proofing. I wanted an expressive enough template to implement everything, and made it a little bit too expressive. Now, it is really for symmetry of the template (which has no value at all implementation-wise).

It can be changed, I have no preference on that.